docs/architecture.md

```
@@ -0,0 +1,451 @@
+# GOTCHA Backend 서버 아키텍처
+
+## 전체 시스템 아키텍처
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)